File: app_java/src/main/java/tech/thdev/android_mvp_sample/presenter/MainPresenter.java
package tech.thdev.android_mvp_sample.presenter;

import android.content.Context;

import java.util.ArrayList;

import tech.thdev.android_mvp_sample.data.ImageItem;
import tech.thdev.android_mvp_sample.data.SampleImageData;

/**
 * Created by tae-hwan on 12/22/16.
 */
public class MainPresenter implements MainContract.Presenter {

    private MainContract.View view;

    private SampleImageData sampleImageData;

    @Override
    public void attachView(MainContract.View view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        view = null;
    }

    @Override
    public void setSampleImageData(SampleImageData sampleImageData) {
        this.sampleImageData = sampleImageData;
    }

    @Override
    public void loadItems(Context context, boolean isClear) {
        ArrayList<ImageItem> items = sampleImageData.getImages(context, 10);
        view.addItems(items, isClear);
        view.notifyAdapter();
    }
}

File: app_kotlin/src/main/java/tech/thdev/app_kotlin/presenter/MainPresenter.kt
package tech.thdev.app_kotlin.presenter

import android.content.Context
import tech.thdev.app_kotlin.data.ImageData

/**
 * Created by tae-hwan on 12/23/16.
 */
class MainPresenter(
    private val view: MainContract.View,
    private val imageData: ImageData
) : MainContract.Presenter {

    override fun loadItems(context: Context, isClear: Boolean) {
        imageData.getSampleList(context, 10).let {
            view.updateItems(it, isClear)
            view.notifyAdapter()
        }
    }
}